framework/BaseClasses/BaseInterface.py

```diff
@@ -0,0 +1,225 @@
+    else:
+      self.raiseAnError(IOError,'not found name for a '+self.__class__.__name__)
+    self.type = paramInput.getName()
+    if self.globalAttributes is not None:
+      self.globalAttributes = globalAttributes
+    if 'verbosity' in paramInput.parameterValues:
+      self.verbosity = paramInput.parameterValues['verbosity'].lower()
+      self.raiseADebug('Set verbosity for '+str(self)+' to '+str(self.verbosity))
+    self._handleInput(paramInput)
+    self.raiseADebug('------Reading Completed for:')
+    self.printMe()
+
+  def _readMoreXML(self,xmlNode):
+    """
+      Function to read the portion of the xml input that belongs to this specialized class
+      and initialize some variables based on the inputs got.
+      @ In, xmlNode, xml.etree.ElementTree.Element, XML element node that represents the portion of the input that belongs to this class
+      @ Out, None
+    """
+    pass
+
+  def _handleInput(self, paramInput):
+    """
+      Function to handle the input parameters that belong to this specialized
+      and initialize variables based on the input.
+      @ In, paramInput, InputData.Parameters
+      @ Out, None
+    """
+    pass
+
+  def whoAreYou(self):
+    """
+      This is a generic interface that will return the type and name of any class that inherits this base class plus all the inherited classes
+      @ In, None
+      @ Out, tempDict, dict, dictionary containing the Type, Class and Name of this instanciated object
+    """
+    tempDict          = {}
+    tempDict['Class'] = '{0:15}'.format(self.__class__.__name__) +' from '+' '.join([str(base) for base in self.__class__.__bases__])
+    tempDict['Type' ] = self.type
+    tempDict['Name' ] = self.name
+    return tempDict
+
+  def getInitParams(self):
+    """
+      Function to be overloaded to get a dictionary of the name and values of the initial parameters associated with any class
+      @ In, None
+      @ Out, paramDict, dict, dictionary containing the parameter names as keys and each parameter's initial value as the dictionary values
+    """
+    return {}
+
+  def myCurrentSetting(self):
+    """
+      This is a generic interface that will return the name and value of the parameters that change during the simulation of any class that inherits this base class.
+      In reality it is just empty and will fill the dictionary calling getCurrentSetting that is the function to be overloaded used as API
+      @ In, None
+      @ Out, paramDict, dict, dictionary containing the current parameters of this instantiated object
+    """
+    paramDict = self.getCurrentSetting()
+    return paramDict
+
+  def getCurrentSetting(self):
+    """
+      Function to be overloaded to inject the name and values of the parameters that might change during the simulation
+      @ In, None
+      @ Out, paramDict, dict, dictionary containing the parameter names as keys and each parameter's initial value as the dictionary values
+    """
+    return {}
```

Comment:
no need to change here but in the future this method will be removed since not used extensively

Reply:
I would welcome that change, it was very odd to determine how this was used and where.